hyperspy/drawing/signal1d.py

```diff
@@ -87,6 +87,11 @@ def add_line(self, line, ax='left'):
             line.sf_lines = self.right_ax_lines
             if line.axes_manager is None:
                 line.axes_manager = self.right_axes_manager
+        if connect_event:
```

Comment:
If this is finally not necessary for this PR (to be discussed) I would suggest to remove it.

Reply:
There are situations, where you don't want to connect the second line (spikes_removal_tool, model line , etc, need to check) to avoid redrawing a second time, the same thing.

Reply:
Do you think it is worth to look for places, when this may be the case and fixes it accordingly?

Reply:
If this feature is gonna stay I would suggest to rename the keyword to something more explicit such as ``update_on_navigation`` or something of this sort.